grpc-testtool/proto/merkle/merkle.proto

```diff
@@ -0,0 +1,101 @@
+syntax = "proto3";
+
+package merkle;
+
+import "google/protobuf/empty.proto";
+
+service Merkle {
+    rpc NewProposal(NewProposalRequest) returns (NewProposalResponse);
+    rpc ProposalCommit(ProposalCommitRequest) returns (ProposalCommitResponse);
+  
+    rpc NewView(NewViewRequest) returns (NewViewResponse);
+    // The methods below may be called with a view ID that corresponds to either a (committable) proposal
+    // or (non-committable) historical view.
+    rpc ViewHas(ViewHasRequest) returns (ViewHasResponse);
+    rpc ViewGet(ViewGetRequest) returns (ViewGetResponse);
+    rpc ViewNewIteratorWithStartAndPrefix(ViewNewIteratorWithStartAndPrefixRequest) returns (ViewNewIteratorWithStartAndPrefixResponse);
+    rpc ViewRelease(ViewReleaseRequest) returns (google.protobuf.Empty);
+}
+
+message NewProposalRequest {
+  // If not given, the parent view is the current database revision.
+  optional uint32 parent_view_id = 1;
+  repeated PutRequest puts = 2;
+  repeated bytes deletes = 3;
+}
+
+message NewProposalResponse {
+  uint32 proposal_id = 1;
+}
+
+message ProposalCommitRequest {
+  uint32 proposal_id = 1;
+}
+
+message ProposalCommitResponse {
+  CommitError err = 1;
+}
+
+message NewViewRequest {
+  bytes root_id = 1;
+}
+
+message NewViewResponse {
+  uint32 view_id = 1;
+}
+
+message ViewHasRequest {
+  uint32 view_id = 1;
+  bytes key = 2;
+}
+
+message ViewHasResponse {
+  bool has = 1;
+}
+
+message ViewGetRequest {
+  uint32 view_id = 1;
+  bytes key = 2;
+}
+
+message ViewGetResponse {
+  bytes value = 1;
+  GetError err = 2;
+}
+
+message ViewNewIteratorWithStartAndPrefixRequest {
```

Comment:
Should this and `ViewNewIteratorWithStartAndPrefixResponse` be here? We don't expose a way to use the iterator once you have its `iterator_id`.

Reply:
Discussed offline